tycho-client/src/feed/mod.rs

```diff
@@ -376,55 +393,76 @@ where
         let mut startup_futures = Vec::new();
         for (id, sh) in sync_streams.iter_mut() {
             let fut = async {
-                let res = timeout(self.block_time, sh.rx.recv()).await;
+                let res = timeout(self.block_time + self.max_wait, sh.rx.recv()).await;
                 (id.clone(), res)
             };
             startup_futures.push(fut);
         }
         let mut ready_sync_msgs = HashMap::new();
-        join_all(startup_futures)
+        let initial_headers = join_all(startup_futures)
             .await
             .into_iter()
-            .for_each(|(extractor_id, res)| {
+            .filter_map(|(extractor_id, res)| {
                 let synchronizer = sync_streams
-                    .get_mut(&extractor_id)
-                    .unwrap();
-                match res {
-                    Ok(Some(msg)) => {
-                        debug!(%extractor_id, height=?&msg.header.number, "Synchronizer started successfully!");
-                        synchronizer.state = SynchronizerState::Ready(msg.header.clone());
-                        ready_sync_msgs.insert(extractor_id.name.clone(), msg);
-                    }
-                    Ok(None) => {
-                        warn!(%extractor_id, "Dead synchronizer at startup will be purged!");
-                        synchronizer.state = SynchronizerState::Ended;
-                    }
-                    Err(_) => {
-                        warn!(%extractor_id, "Stale synchronizer at startup will be purged!");
-                        synchronizer.state = SynchronizerState::Ended;
-                    }
+                .get_mut(&extractor_id)
+                .unwrap();
+            match res {
+                Ok(Some(msg)) => {
+                    debug!(%extractor_id, height=?&msg.header.number, "Synchronizer started successfully!");
+                    // initially default all synchronizers to Ready
+                    synchronizer.state = SynchronizerState::Ready(msg.header.clone());
+                    synchronizer.modify_ts = Local::now().naive_utc();
+                    ready_sync_msgs.insert(extractor_id.name.clone(), msg.clone());
+                    Some(msg.header)
                 }
-            });
+                Ok(None) => {
+                    warn!(%extractor_id, "Dead synchronizer at startup will be purged!");
+                    synchronizer.state = SynchronizerState::Ended;
+                    synchronizer.modify_ts = Local::now().naive_utc();
+                    None
+                }
+                Err(_) => {
+                    warn!(%extractor_id, "Stale synchronizer at startup will be purged!");
```

Comment:
So the error here is if we timed out waiting for a message, right? Maybe this can be made a bit clearer in the warning message?